lib/hammer_cli_foreman_tasks/command_extensions/recurring_logic.rb

```diff
@@ -0,0 +1,54 @@
+require 'date'
+
+module HammerCLIForemanTasks
+  module CommandExtensions
+    class RecurringLogic < HammerCLI::CommandExtensions
+      before_print do |data|
+        data['action'] = format_task_input(data['tasks'].last)
+        data['last_occurrence'] = recurring_logic_last_occurrence(data)
+        data['next_occurrence'] = recurring_logic_next_occurrence(data)
+        data['iteration_limit'] = format_recurring_logic_limit(data['max_iteration'])
+        data['repeat_until'] = format_recurring_logic_limit(data['end_time'])
+      end
+
+      output do |definition|
+        definition.insert(:after, :cron_line) do
+          field :action, _('Action')
+          field :last_occurrence, _('Last occurrence')
+          field :next_occurrence, _('Next occurrence')
+        end
+        definition.insert(:after, :iteration) do
+          field :iteration_limit, _('Iteration limit')
+        end
+        definition.insert(:replace, :end_time) do
+          field :repeat_until, _('Repeat until')
+        end
+      end
+
+      def self.recurring_logic_last_occurrence(recurring_logic)
+        last_task = recurring_logic['tasks'].max { |a, b| a['start_at'] <=> b['start_at'] }
+        return '-' if last_task.nil? || last_task['started_at'].nil?
```

Comment:
```
$ hammer -d recurring-logic info --id 6

-----B<-----SNIP-----B<-----

[DEBUG 2020-05-05T14:52:00 API] Response: {
               "id" => 6,
        "cron_line" => "13 16 * * *",
         "end_time" => "2021-05-04 11:13:00 UTC",
        "iteration" => 2,
    "task_group_id" => 159,
            "state" => "active",
    "max_iteration" => 10,
            "tasks" => [
        [0] {
                           "id" => "2bf5df4e-e3a8-44b0-8137-f44032e477b6",
                        "label" => "Actions::RemoteExecution::RunHostsJob",
                      "pending" => false,
                       "action" => "Run hosts job: Run ls -la /",
                     "username" => "admin",
                   "started_at" => "2020-05-04 14:13:10 UTC",
                     "ended_at" => "2020-05-04 14:13:22 UTC",
                        "state" => "stopped",
                       "result" => "success",
                     "progress" => 1.0,
                        "input" => {
                         "job_invocation" => {
                             "id" => 153,
                           "name" => "Commands",
                    "description" => "Run ls -la /"
                },
                     "current_request_id" => "ba159fb9-c0f9-426a-97c2-df279d3f2e6b",
                       "current_timezone" => "Europe/Prague",
                        "current_user_id" => 4,
                "current_organization_id" => 1,
                    "current_location_id" => 2,
                           "job_category" => "Commands",
                      "job_invocation_id" => 153
            },
                       "output" => {
                  "planned_count" => 1,
                "cancelled_count" => 0,
                    "total_count" => 1,
                   "failed_count" => 0,
                  "pending_count" => 0,
                  "success_count" => 1
            },
                    "humanized" => {
                "action" => "Run hosts job:",
                 "input" => "Run ls -la /",
                "output" => "1 task(s), 1 success, 0 fail",
                "errors" => []
            },
                  "cli_example" => nil,
                     "start_at" => "2020-05-04 14:13:00 UTC",
            "available_actions" => {
                "cancellable" => false,
                  "resumable" => false
            }
        },
    [1] {
                           "id" => "c4c3aa9e-2bef-42b2-87b6-da7c879994dc",
                        "label" => "Actions::RemoteExecution::RunHostsJob",
                      "pending" => true,
                       "action" => "Run hosts job:",
                     "username" => "admin",
                   "started_at" => nil,
                     "ended_at" => nil,
                        "state" => "scheduled",
                       "result" => "pending",
                     "progress" => 0,
                        "input" => {
                         "job_invocation" => {
                             "id" => 157,
                           "name" => "Commands",
                    "description" => "Run ls -la /"
                },
                     "current_request_id" => "b74a1240-1a69-46a0-8904-5a987ff818b8",
                       "current_timezone" => "Europe/Prague",
                        "current_user_id" => 4,
                "current_organization_id" => 1,
                    "current_location_id" => 2
            },
                       "output" => {},
                    "humanized" => {
                "action" => "Run hosts job:",
                 "input" => nil,
                "output" => nil,
                "errors" => nil
            },
                  "cli_example" => nil,
                     "start_at" => "2020-05-05 14:13:00 UTC",
            "available_actions" => {
                "cancellable" => true,
                  "resumable" => false
            }
        }
    ]
}

-----B<-----SNIP-----B<-----

ID:              6                                                       
Cron line:       13 16 * * *                        
Action:          Run hosts job:            
Last occurrence: -                     
Next occurrence: 2020-05-05 14:13:00 UTC
Iteration:       2                                                                                                                                            
Iteration limit: 10                                                                                                                                           
Repeat until:    2021-05-04 11:13:00 UTC   
State:           active
```

I don't think this is correct. Shouldn't it sort by `started_at` to avoid finding the "latest" task (which possibly wasn't started yet) and then showing `-` as last occurrence time?

Reply:
Yes, thanks, you're right. While refactored I've misspelled somehow. Should be fixed now.